src/cashlib/cashlib.cpp

```diff
@@ -845,7 +868,13 @@ extern "C" JNIEXPORT jstring JNICALL Java_bitcoinunlimited_libbitcoincash_PayAdd
 
     env->ReleaseByteArrayElements(arg, data, 0);
 
-    std::string addrAsStr(EncodeCashAddr(dst, *cashlibParams));
+    const CChainParams *cp = GetChainParams((ChainSelector)chainSelector);
+    if (cp == nullptr)
+    {
+        throwIllegalState(env, "Unknown blockchain selection");
+        return nullptr;
+    }
```

Comment:
if you're throwing here do you really need the return statement?  I think there's another one below.

Reply:
Yeah, I put them in to avoid the warning but shouldn't happen in this case, will fix

Reply:
Actually, this function is "throwing" a java exception, which just sets some program state.  Its not throwing in a C++ sense, so the return is still needed

Reply:
changed the name for clarity